x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -726,7 +758,13 @@ public Object visitInputNamedOrPositionalParam(EsqlBaseParser.InputNamedOrPositi
     private Object visitParam(EsqlBaseParser.ParamsContext ctx, QueryParam param) {
         Source source = source(ctx);
         DataType type = param.type();
-        return new Literal(source, param.value(), type);
+        if (param.isPattern()) {
+            return new UnresolvedAttribute(source, (String) param.value(), "Field name pattern [" + param.value() + "] is not supported");
```

Comment:
use `UnresolvedNamePattern` instead of `UnresolvedAttribute`

Reply:
Done.